src/WebJobs.Script/Host/ScriptHost.cs

```diff
@@ -465,31 +467,40 @@ private async Task InitializeWorkersAsync()
                     server.Uri,
                     _loggerFactory); // TODO: DI (FACAVAL) Pass appropriate logger. Channel facory should likely be a service.
             };
-            var configFactory = new WorkerConfigFactory(ScriptSettingsManager.Instance.Configuration, _startupLogger);
+
+            _functionDispatcher = new FunctionDispatcher(EventManager, server, channelFactory, _workerConfigs);
+
+            _eventSubscriptions.Add(EventManager.OfType<WorkerProcessErrorEvent>()
+                .Subscribe(evt =>
+                {
+                    HandleHostError(evt.Exception);
+                }));
+        }
+
+        internal static IEnumerable<WorkerConfig> GetWorkerConfigs(string language, ILogger logger)
```

Comment:
Can we move this out of `ScriptHost`? We're trying to reduce the mixed logic in this class.